dhnx/optimization/precalc_hydraulic.py

```diff
@@ -25,48 +25,345 @@
+    # für argumente siehe https://stackoverflow.com/questions/19843116/passing-arguments-to-fsolve
+
+
+def eq_transition(x, *data):
+    r"""
+    Equation to be solved for the transition range.
+
+    Equation
+    --------
+    .. eq_transition_equation
+
+    :math:`f(x)=x+2\cdot log \Bigg(  \frac{2,51x}{Re} \cdot \frac{k}{3,71d_i} \Bigg)`
+
+    Parameters
+    ----------
+    x : numeric
+        :math:`x`: function variable
+
+    R_e: numeric
+        :math:`Re`: Reynolds number
+
+    k : numeric
+        :math:`k`: roughness of inner pipeline surface [mm]
+
+    di : numeric
+        :math:`d_i`: inner diameter [m]
+
+    Returns
+    -------
+    Equation : numeric
+
+    """
+    R_e, k, d_i = data  # siehe oben
+
+    return x + 2 * np.log10((2.51 * x) / R_e + k / (3.71 * d_i))
+
+
+def calc_k_v(d_v, d):
+    r"""
+    Calcutlates the kinematic viscosity for given density and dynamic viscosity
+
+    Formula
+    -------
+    .. calc_k_v_equation:
+
+    :math:`\nu = \frac{\eta}{\rho}`
+
+    Parameters
+    ----------
+    d_v: numeric
+        :math:`\eta`: dynamic viskosity [kg/(m*s)]
+
+    d: numeric
+        :math:`\rho`: density [kg/m³]
+
+    Returns
+    -------
+    kinematic viscosity [m²/s] : numeric
+
+    """
+    return d_v / d
+
+
+def calc_Re(v, d_i, k_v):
+    r"""
+    Calculates the Reynolds number for a given velocity, inner diameter and kinematic viscosity
+
+    Formula
+    -------
+    .. calc_Re_equation:
+
+    :math:`\frac{v \cdot d_i}{\nu}`
+
+    Parameters
+    ----------
+    v: numeric
+        :math:`v`: flow velocity [m/s]
+
+    d_i: numeric
+        :math:`d_i`: inner pipe diameter [m]
+
+    k_v: numeric
+        :math:`\nu`: kinematic viscosity [m²/s]
+
+    Returns
+    -------
+    Reynolds number []: numeric
+
+    """
+    return v * d_i / k_v
+
+
+def calc_lam_lam(Re):
+    r"""
+    Calculates the Darcy friction factor for a given Reynolds number for a laminar flow.
+
+    Formula
+    -------
+    .. calc_lam_lam_equation:
+
+    :math:`\lambda=\frac{64}{Re}`
+
+    Parameters
+    ----------
+    Re: numeric
+        :math:`Re`: Reynolds number
+
+    Returns
+    -------
+    Darcy friction factor [] : numeric
+
+    """
+    return 64 / Re
+
+
+def calc_d_p(lam, length, d_i, d, v):
+    r"""
+    Calculates the pressure drop in a pipe for a given Darcy friction factor.
+
+    Formula
+    -------
+    .. calc_d_p_equation:
+
+    :math:`\Delta p = \lambda \frac{l}{d_i} \frac{\rho}{2} v^2`
+
+    Parameters
+    ----------
+    lam: numeric
+        :math:`\lambda`: Darcy friction factor []
+
+    length: numeric
+        :math:`l`: length of the pipe [m]
+
+    d_i : numeric
+        :math:`d_i`: inner pipe diameter [m]
+
+    d: numeric
+        :math:`\rho`: density [kg/m³]
+
+    v: numeric
+        :math:`v`: flow velocity [m/s]
+
+    Returns
+    -------
+    Pressure drop [Pa]: numeric
+    """
+    return lam * length / d_i * d / 2 * v ** 2
+
+
+def calc_lam_turb1(Re):
```

Comment:
ah, now I think I get it. Maybe, it should be *laminar_smooth*, *turbulent_rough* *turb-smooth* and so on ... ?